tests/func/test_checkout.py

```diff
@@ -68,6 +68,37 @@ def test(self):
         self._test_checkout()
 
 
+class TestCheckoutPartial(TestRepro):
+
+    CO_DIR = "checkout_partial"
+
+    def test(self):
+
+        files = [
+            os.path.join(self.CO_DIR, filename)
+            for filename in ("a.txt", "b.txt")
+        ]
+        for file in files:
+            self.create(file, file)
+
+        stages = self.dvc.add(self.CO_DIR)
+        self.assertEqual(len(stages), 1)
+
+        for file in files:
+            os.unlink(file)
+        os.rmdir(self.CO_DIR)
+
+        file1, file2 = files
+
+        self.dvc.checkout(targets=file1)
+        self.assertTrue(os.path.exists(self.CO_DIR))
+        self.assertTrue(os.path.exists(file1))
+        self.assertFalse(os.path.exists(file2))
+
+        self.dvc.checkout(targets=file2)
+        self.assertTrue(os.path.exists(file2))
```

Comment:
We are currently continuing to migrate from unittest-style tests to pytest-style tests https://github.com/iterative/dvc/issues/1819 , so all new tests need to be in pytest-style. Could you please rewrite it? E.g. see examples at the bottom of this test_checkout.py file.

Reply:
Also, we need to test a case for checking out a subdirectory in a tracked directory, that might break.

Reply:
Applied both of them